nucypher/blockchain/eth/domains.py

```diff
@@ -25,10 +25,17 @@ class PolygonChain(ChainInfo, Enum):
 
 
 class TACoDomain:
-    def __init__(self, name: str, eth_chain: EthChain, polygon_chain: PolygonChain):
+    def __init__(
+        self,
+        name: str,
+        eth_chain: EthChain,
+        polygon_chain: PolygonChain,
+        condition_chains: Tuple[ChainInfo, ...],
```

Comment:
Also not for this PR, but we shouldn't make a distinction between types of chains/providers. I'm thinking there should be some sort of class abstracting a pool of endpoints, potentially performing all validation checks needed (liveness, chain_id is ok, chain_id is valid for conditions, etc)